src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -285,22 +282,8 @@ bool InputStateMachineEngine::ActionPrintString(const std::wstring_view string)
 // - flush - not applicable to the input state machine.
 // Return Value:
 // - true iff we successfully dispatched the sequence.
-bool InputStateMachineEngine::ActionPassThroughString(const std::wstring_view string, const bool /*flush*/)
+bool InputStateMachineEngine::ActionPassThroughString(const std::wstring_view string)
 {
-    if (_pDispatch->IsVtInputEnabled())
-    {
-        // Synthesize string into key events that we'll write to the buffer
-        // similar to TerminalInput::_SendInputSequence
-        if (!string.empty())
-        {
-            InputEventQueue inputEvents;
-            for (const auto& wch : string)
-            {
-                inputEvents.push_back(SynthesizeKeyEvent(true, 1, 0, 0, wch, 0));
-            }
-            return _pDispatch->WriteInput(inputEvents);
-        }
-    }

```

Comment:
Don't we need the thing that converts passed-through VT content into raw synthesized key events?

e.g. is there a difference between how keys through `_pDispatch->WriteString()` are synthesized compared to this (where it's just a raw wch for VtInput consumers)?

Reply:
fwiw, InteractDispatch uses `CharToKeyEvents`, which makes real-looking key events... unlike the synthetic ones we create for VT output. Like, it will include a scancode for ESC, and both the keydown and keyup.